l1-contracts/contracts/bridgehub/ChainAssetHandler.sol

```diff
@@ -121,22 +187,34 @@ contract ChainAssetHandler is
         returns (bytes memory bridgehubMintData)
     {
         BridgehubBurnCTMAssetData memory bridgehubBurnData = abi.decode(_data, (BridgehubBurnCTMAssetData));
-        if (_assetId != BRIDGEHUB.ctmAssetIdFromChainId(bridgehubBurnData.chainId)) {
-            revert IncorrectChainAssetId(_assetId, BRIDGEHUB.ctmAssetIdFromChainId(bridgehubBurnData.chainId));
+        require(
+            _assetId == BRIDGE_HUB.ctmAssetIdFromChainId(bridgehubBurnData.chainId),
+            IncorrectChainAssetId(_assetId, BRIDGE_HUB.ctmAssetIdFromChainId(bridgehubBurnData.chainId))
+        );
+        address zkChain = BRIDGE_HUB.getZKChain(bridgehubBurnData.chainId);
+
+        /// We set the isL1ToL2DepositProcessed flag on the asset tracker to demarcate deposits happening before and after the migration.
+        /// We set the legacy shared bridge address on the gateway asset tracker to allow for L2->L1 asset withdrawals via the L2AssetRouter.
+        if (block.chainid == L1_CHAIN_ID) {
+            bytes memory data = abi.encodeCall(
```

Comment:
If a chain is already settling on top of GW at the time of the upgrade, what would happen to this value?

On another hand, the set of chains that have this legacy address is very finite, maybe it makes sense to just hardcode it somewhere? To ensure that this value will be set automatically for these chains during v30 upgrade. We can make it in a separate PR